### PR TITLE
[ROCm] add pytorch-triton-rocm wheel build

### PR DESCRIPTION
**Pull request by jeffdaily**
Add workflow for building triton-rocm wheel.

cc @malfet @seemethere @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport

### CODE REVIEW

**Comment by atalman** [comment]
cc @malfet

File: setup.py [hunk 1]
```diff
@@ -1022,11 +1022,19 @@ def main():
...
+            if os.path.exists(triton_pin_file):
+                with open(triton_pin_file) as f:
+                    triton_pin = f.read().strip()
+                    extras_require['dynamo'] = ['pytorch-triton-rocm==2.0.0+' + triton_pin[:10], 'jinja2']
+        else:
+            triton_pin_file = os.path.join(cwd, ".github", "ci_commit_pins", "triton.txt")
+            if os.path.exists(triton_pin_file):
+                with open(triton_pin_file) as f:
+                    triton_pin = f.read().strip()
+                    extras_require['dynamo'] = ['pytorch-triton==2.0.0+' + triton_pin[:10], 'jinja2']
```
Comment (malfet) [inline setup.py:1037]: Plz, avoid code duplication
```suggestion
       pin_name, pkg_name = ("pytorch-trition", "trition.txt") is not os.path.exists("/opt/rocm") else ("pytorch-trition-rocm", "trition-rocm.txt")
        triton_pin_file = os.path.join(cwd, ".github", "ci_commit_pins", pin_name)
        if os.path.exists(triton_pin_file):
            with open(triton_pin_file) as f:
                triton_pin = f.read().strip()
                extras_require['dynamo'] = [f'{pkg_name}==2.0.0+' + triton_pin[:10], 'jinja2']
```

File: .github/workflows/build-triton-rocm-wheel.yml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_S3_UPDATE_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_S3_UPDATE_SECRET_ACCESS_KEY }}
+          UPLOAD_BUCKET: "s3://pytorch"
+        run: |
+            set -ex
+            pip install -q awscli
+            s3_dir="${UPLOAD_BUCKET}/whl/nightly/"
+            for pkg in "${PKG_DIR}/"*.whl; do
+              aws s3 cp --no-progress --acl public-read "${pkg}" "${s3_dir}"
+             done
```
Comment (malfet) [inline .github/workflows/build-triton-rocm-wheel.yml:148]: Can we reuse the same workflow plz, rather than copy-n-paste?

**Comment by jeffdaily** [comment]
Closing.  We will host wheels on pypi for pytorch-triton-rocm for PT2 release.

https://pypi.org/project/pytorch-triton-rocm/#history

**Pull request closed without merging**